src/main/java/tasks/TaskList.java

```diff
@@ -0,0 +1,156 @@
+
+import java.util.ArrayList;
+
+/**
+ * A class that abstracts a list of tasks set by the user. It also contains methods
+ * that pertain to updating the list of tasks.
+ */
+public class TaskList {
+
+    /** An arraylist that contains the tasks set by the user. */
+    private final ArrayList<Task> taskArrayList = new ArrayList<>();
+    /** The number of uncompleted tasks in the task list */
+    private int uncompletedTasks = 0;
+    /** The total number of tasks in the task list. */
+    private int totalTasks = 0;
+
+    /**
+     * Adds a task to the task list. Prints a message to tell the user that the task has been added.
+     *
+     * @param task The task to be added to the task list.
+     */
+    public String addTask(Task task) {
+        String message;
+        this.taskArrayList.add(task);
+        this.uncompletedTasks++;
+        totalTasks++;
+        message = "Got it! This task has been added:\n" + task + "\n" + this.getTaskListStatus();
+        this.saveTaskList();
+        return message;
+    }
+
+    /**
+     * Adds a previously saved task to the task list. This method should only be used when loading a task from
+     * memory. No message is shown to the user when a task is added. Do not use this method to add a newly
+     * inputted task from the user to the taskList.
+     *
+     * @param task The previously saved task to add to the taskList.
+     */
+    public void addSavedTask(Task task) {
+        this.taskArrayList.add(task);
+        if (!task.isDone()) {
+            this.uncompletedTasks++;
+        }
+        this.totalTasks++;
+    }
+
+    /**
+     * Removes a task in the task list based on the index given.
+     *
+     * @param index The index of the task list. Note that the index provided starts from 1. So the
+     *              index 1 represents the first task in the taskList ArrayList.
+     * @return The message that should be shown to the user after removing a task.
+     */
+    public String removeTask(int index) {
+        String message;
+        if (index > this.taskArrayList.size() || index <= 0) {
+            message = "A task could not be removed because it does not exist.\n";
+            return message;
+        }
+        Task task = this.taskArrayList.get(index - 1);
+        this.taskArrayList.remove(index - 1);
+        if (!task.isDone()) {
+            uncompletedTasks--;
+        }
+        totalTasks--;
+        message = "Understood. I've removed this task:\n" + task + "\n";
+        this.saveTaskList();
+        return message;
+    }
+
+    /**
+     * Marks a task in the task list as completed based on the index given.
+     *
+     * @param index The index of the task list. Note that the index provided starts from 1.
+     *              So the index 1 represents the first task in the taskList ArrayList.
+     * @return The message that should be shown to the user after a task has been marked as done.
+     */
+    public String markTaskAsCompleted(int index) {
```

Comment:
Perhaps the terminology for a completed task could be standardised? Currently there's a few terms used to refer to this: "finished", "done", "completed".